### PR TITLE
New UPP guns

### PR DESCRIPTION
**Pull request by CapCamIII**
# About the pull request

Adds a bunch of new UPP guns to replace ill-fitting ones, respites others, removes others, also does some balance changes that I feel would be good for the game, nothing too outrageous I believe.

**All sprites by wei/esselnek** 

[lore and stuff](https://docs.google.com/document/d/11eKNeVflsxy57fFg1GcwNZk9llqRnnlJolEU36MBEh0/edit?usp=sharing)

Look at changelog and code for specific changes unless i get asked by a dev to list them here

# Explain why it's good for the game

New UPP guns to replace old things that don't make sense, better sprites, IMO needed balance changes.

# Testing Photographs and Procedure
image
Slightly outdated and I forgot Type 23, but mostly correct. Look at imagediff for type 23 sprite

# Changelog
I might have forgotten a thing or two since I've been working on this on and off since April, but from what I can tell mostly there

:cl: 
add: Adds 6(8 if you're pedantic) new guns for the UPP, details for each in following lines.
add: Type 19, Replacement for PPsh on colonies and black market.
add: Type 64, replacement for scorpion, full sized SMG, 2 versions, 1 for upp, 1 for not upp
add: HJRA-12, RPG for UPP.
add: Type 73, pistol for higher ranked UPP troops, UPP COs get a Type 74 which has high-impact rounds, replaces PK9 in non-UPP use.
add: NP92, pistol, replaces PK9 in UPP use(mostly), integrally suppressed version called NPZ92 for UPP Commandos.
add: QYJ-72, UPP GPMG, replaces minigun with UPP Minigunners
add: Technically resprites but due to changes closer to replaces N-Y revolver with ZHNK-72
add: Adds a character custom loadout Type 73, 2 mags spare.
add: UPP helmets now work as ear protection, intended use with HJRA-12 RPG
add: UPP officers now have regular UM4 helmets instead of UH7 helmets in their vendors as UH7 looked ugly with their combat gear
del: Removes the PK9 entirely
del: Removes tranqs from UPP Commando loadout
del: Removes HEAP ammo from UPP officers and commandos, replaced with normal and AP mags when applicable
balance: UPP jackets no longer have massive armor, in exchange they no longer slow and in clothing vendors a light version of UPP armor is vendable
balance: UPP MPs are now armed with a ZHNK-72 belt on their jacket, with an additional shrapnelshot loader in their belt.
balance: Rebalances Type 71 damage/firerate/AP to be a sidegrade of M41A, about equal DPS, higher damage per shot and AP, lower fire rate, carbine is round-about the exact same as M41A however.
balance: Fixes UPP support armor, given to medics on their person and officers in their vendor, lighter version of UPP armor, exchanges protection for speed.
balance: Seperates SVD bullet damage from crude rifle bullet and buffs it as a hard-hitting AP bullet.
balance: Rebalances UPP revolver(now ZHNK-72), much harder hitting and has higher AP, but rarer, shrapnelshot extremely rare and good.
balance: Due to ammo changes, SVD no longer has burst.
balance: Removes ceramic plates from UPP loadouts
fix: UPP Major now has the correct UPP Commander skillset
fix: UPP Commander skillset can BE
fix: Fixes typepath for UPP support/light armor, actually works now.
spellcheck: Renames SVD to Type 88
spellcheck: Renames UPP Minigunner to UPP Machinegunner
spellcheck: Renames UPP jackets to UL4 all around
soundadd: Adds firing sounds for Type 73 and SVD/UPP MG
imageadd: Resprites the SVD, Type 71, and Type 23, all sprites by wei/esselnek
imageadd: Adds sprites for all above guns, all sprites by wei/esselnek
maptweak: Changes instances of skorpion to non-upp type 64, also changes instances of ppsh to type 19
/:cl:

### CODE REVIEW

**Comment by BeagleGaming1** [comment]
Could you elaborate on what this was?
> qol: made something easier to use

**Comment by CapCamIII** [comment]
> Could you elaborate on what this was?
> 
> > qol: made something easier to use

I forgot to remove the default CL for QOL

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by c4xmaniac2** [comment]
holy shit this looks good

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cuberound** [comment]
I think there will be issue if this gets merged without changes in UPP loadout in datums/module/equipment_preset/ . Or the ERT will call bunch of nonexisting guns on spawn. Can be done in separate pull reqest

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by nauticall — APPROVED** [review]
these look so much nicer. sprites approved, lets get this IN

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: maps/map_files/Ice_Colony_v3/Shivas_Snowball.dmm
Comment (Nanu308) [inline maps/map_files/Ice_Colony_v3/Shivas_Snowball.dmm:1]: Mapping lgtm

File: maps/map_files/BigRed/BigRed.dmm
Comment (Nanu308) [inline maps/map_files/BigRed/BigRed.dmm:1]: Mapping lgtm

File: maps/map_files/Sorokyne_Strata/Sorokyne_Strata.dmm
Comment (Nanu308) [inline maps/map_files/Sorokyne_Strata/Sorokyne_Strata.dmm:1]: Mapping lgtm

**Review by Nanu308 — APPROVED** [review]
mapping lgtm

File: code/modules/projectiles/guns/rifles.dm [hunk 1]
```diff
@@ -1436,47 +1465,45 @@
...
+	SUPPRESSOR.flags_attach_features &= ~ATTACH_REMOVABLE
+	SUPPRESSOR.Attach(src)
+	update_attachable(SUPPRESSOR.slot)
 	//scope
-	var/obj/item/attachable/scope/mini/F = new(src)
-	F.hidden = TRUE
-	F.flags_attach_features &= ~ATTACH_REMOVABLE
-	F.Attach(src)
-	update_attachable(F.slot)
+	var/obj/item/attachable/scope/mini/SCOPE = new(src)
```
Comment (morrowwolf) [inline code/modules/projectiles/guns/rifles.dm:1498]: No all caps variable names

File: code/modules/projectiles/guns/pistols.dm [hunk 2]
```diff
@@ -258,66 +258,132 @@
...
-	item_state = "pk9u"
 
-/obj/item/weapon/gun/pistol/c99/upp/tranq
-	desc = "The Korovin PK-9 is a cheap, robust and reliable sidearm, its design strongly inspired by the classic ancient Makarov pistol. This version contains a customized exterior, an integrated laser and reflex sight, and is noticeably easy to handle."
-	icon_state = "pk9r"
-	item_state = "pk9r"
-	current_mag = /obj/item/ammo_magazine/pistol/c99/tranq
-	aim_slowdown = 0
-	wield_delay = WIELD_DELAY_MIN
+/obj/item/weapon/gun/pistol/t73/leader
```
Comment (morrowwolf) [inline code/modules/projectiles/guns/pistols.dm:349]: If this has high impact rounds it needs to have a far more distinctive firing sound.

**Review by morrowwolf — CHANGES_REQUESTED** [review]
No plates.

Do another pass over the black market changes, if I find something stupid in there in like a month I'm gonna be greatly annoyed.

**Review by Zonespace27 — APPROVED** [review]

**Pull request merged**